backend/src/tasks/trustymail.ts

```diff
@@ -0,0 +1,120 @@
+import getAllDomains from './helpers/getAllDomains';
+import { ChildProcessWithoutNullStreams, spawn } from 'child_process';
+import { CommandOptions } from './ecs-client';
+import saveTrustymailResultsToDb from './helpers/saveTrustymailResultsToDb';
+import * as chokidar from 'chokidar';
+
+// TODO: Add scan time to domain table
+// TODO: Retry failed domains
+// TODO: PSL is downloaded once per organization per scan; would like to download only once per day
+export const handler = async (commandOptions: CommandOptions) => {
+  let pslDownloaded = false;
+  const { organizationId, organizationName } = commandOptions;
+  console.log('Running Trustymail on organization', organizationName);
+  const domains = await getAllDomains([organizationId!]);
+  const queue = new Set(domains.map((domain) => `${domain.id}.json`));
+  const failedDomains = new Set<string>();
+  // Event listener detects when public_suffix_list.dat is downloaded
+  chokidar.watch('public_suffix_list.dat').on('add', () => {
+    console.log('Public suffix list downloaded');
+    pslDownloaded = true;
+  });
+  console.log(
+    `${organizationName} domains to be scanned:`,
+    domains.map((domain) => domain.name)
+  );
+  console.log(`queue`, queue);
+  for (const domain of domains) {
+    const path = `${domain.id}.json`;
+    // Event listener detects Trustymail results file at creation and syncs it to db
+    chokidar.watch(path).on('add', (path) => {
```

Comment:
Do we need to use chokidar to watch to wait for the file to be created at this path? Using chokidar feels a bit clunky / more complex / might be more buggy. What if we did either of the following:
1. called `spawnSync` instead of `spawn`, then waited for the process to finish, then read the `public_suffix_list.dat` file?
2. called `spawn`, and listen for it to finish `child.on('exit')` and then read the `public_suffix_list.dat` file?

That would be much simpler, I think.

Reply:
It would be good if you could update this in this PR. But let me know if there might be another issue I'm overlooking?

Reply:
I agree that waiting for the PSL is clunky and I'm exploring some alternatives. I initially used spawnSync as you suggest but scans take 3-4 times longer. And with child.on('exit') scans were running so quickly that several calls to download the PSL happened before the first process exited; my workaround was coding a delay between each scan so this was also a little slower. That said, maybe you could help me get one of these methods to work more efficiently.

What I'm looking into now is adding code to worker.ts that would follow the logic:
`if (scanName === 'trustymail') { update PSL file in backend/app}`
The intent would be that the spawned containers would already have a fresh copy of the PSL and I could remove the logic to download it in trustymail.ts.

Reply:
I removed the event listener for public_suffix_list.dat. PSL is now checked/updated in Dockerfile.worker and worker.ts to compensate.